### PR TITLE
Creating bazel toolchain to crosscompile with

### PR DESCRIPTION
**Pull request by deannagarcia**
Creating a bazel toolchain to cross compile to all architectures supported with Linux.

### CODE REVIEW

File: toolchain/cc_toolchain_config.bzl [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  return cc_common.create_cc_toolchain_config_info(
+      abi_libc_version = ctx.attr.target_cpu,
+      abi_version = ctx.attr.target_cpu,
+      ctx = ctx,
+      compiler = "clang",
+      cxx_builtin_include_directories = [
+          ctx.attr.toolchain_dir,
+          "/usr/include",
+          "/usr/lib/clang/11.0.1/include",
```
Comment (perezd) [inline toolchain/cc_toolchain_config.bzl:129]: Is there potentially a symlink somewhere that lets us avoid using explicit versions?

File: toolchain/BUILD [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    toolchains = {
+        "aarch64": ":cc-compiler-aarch64",
+        "ppcle": "cc-compiler-ppcle",
+        "s390x": ":cc-compiler-s390x",
+        "x86_32": ":cc-compiler-x86_32",
+        "x86_64": ":cc-compiler-x86_64",
+    },
+)
+
+CC_TOOLCHAIN_CPUS = [
```
Comment (perezd) [inline toolchain/BUILD:18]: Would it be possible to condense the toolchains dictionary on L9 and this into a single variable that can be reused in the two places? That way you'd be able to use cpu + name from the dictionary directly in the for loop of cc_toolchain, and you'd have less duplication here.

File: toolchain/BUILD [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        output_licenses = ["restricted"],
+        static_runtime_lib = ":empty",
+        strip_files = ":empty",
+        toolchain_config = ":" + cpu + "-config",
+        toolchain_identifier = "linux_" + cpu,
+    )
+    for cpu in CC_TOOLCHAIN_CPUS
+]
+
+cc_toolchain_config(
```
Comment (perezd) [inline toolchain/BUILD:44]: It'd be good to prefix all the name fields with the OS here as well so they don't collide potentially with other OSes.

**Review by perezd — CHANGES_REQUESTED** [review]
Looking good! Nice work.

File: toolchain/cc_toolchain_config.bzl (hunk 1 above)
Comment (deannagarcia) [reply toolchain/cc_toolchain_config.bzl]: I removed the version and it still works.

File: toolchain/BUILD (hunk 2 above)
Comment (deannagarcia) [reply toolchain/BUILD]: Done!

File: toolchain/BUILD (hunk 3 above)
Comment (deannagarcia) [reply toolchain/BUILD:37]: Done!

File: toolchain/cc_toolchain_config.bzl [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                              "--target=" + ctx.attr.target_full_name,
+                          ],
+                      ),
+                  ]),
+              ),
+          ],
+      ),
+  ]
+
+  features.append(feature(
```
Comment (perezd) [inline toolchain/cc_toolchain_config.bzl:98]: any reason to not just inline this as part of the array literal? vs calling append?

File: toolchain/BUILD [hunk 5]
```diff
@@ -0,0 +1,83 @@
+load(":cc_toolchain_config.bzl", "cc_toolchain_config")
+package(default_visibility = ["//visibility:public"])
```
Comment (perezd) [inline toolchain/BUILD:2]: ultra nit: add space between load and package.

**Review by perezd — APPROVED** [review]
LGTM!

File: toolchain/cc_toolchain_config.bzl (hunk 4 above)
Comment (deannagarcia) [reply toolchain/cc_toolchain_config.bzl]: Nope, added it to part of the array literal.

File: toolchain/BUILD (hunk 5 above)
Comment (deannagarcia) [reply toolchain/BUILD:3]: Done!

**Pull request merged**